src/Servers/HttpSys/src/FeatureContext.cs

```diff
@@ -635,6 +635,13 @@ private void ConsiderEnablingResponseCache()
         private static TimeSpan? GetCacheTtl(RequestContext requestContext)
         {
             var response = requestContext.Response;
+            // A 304 response is supposed to have the same headers as its associated 200 response, including Cache-Control, but the 304 response itself
+            // should not be cached. Otherwise Http.Sys will serve the 304 response to all requests without checking conditional headers like If-None-Match.
+            if (response.StatusCode == StatusCodes.Status304NotModified)
```

Comment:
So HTTP.Sys doesn't use the If-None-Match header as part of the cache lookup key. Can that be configured somehow?

Reply:
No, and no.